src/lib/mdns/Scanner.h

```diff
@@ -0,0 +1,48 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "core/CHIPError.h"
+#include "inet/IPAddress.h"
+#include "inet/InetInterface.h"
+
+namespace chip {
+namespace Mdns {
+
+class ScannerDelegate
+{
+public:
+    virtual void HandleNodeResolved(CHIP_ERROR error, uint64_t nodeId, uint64_t fabricId, const Inet::IPAddress & address) = 0;
+    virtual ~ScannerDelegate() {}
+};
+
+class Scanner
+{
+public:
+    virtual ~Scanner() {}
+
+    virtual CHIP_ERROR RegisterScannerDelegate(ScannerDelegate * delegate)                       = 0;
+    virtual CHIP_ERROR SubscribeNode(uint64_t nodeId, uint64_t fabricId, Inet::InterfaceId interface = INET_NULL_INTERFACEID,
```

Comment:
Please add some documentation to this: what does each method do?

Do we need a method to 'initiate mDNS scan now' ? I would expect CHIP devices to passively listen for bootup advertisements when we register to listen for a node, however I should also be able to trigger a 'find node now' when needed.

Alternatives seem to be to just periodically scan, however that does not seem friendly to the network traffic, especially as number of chip devices increases.

Reply:
Added docs and a new method to trigger an instant scan.